blobfuse/utilities.cpp

```diff
@@ -19,6 +19,108 @@ std::string prepend_mnt_path_string(const std::string path)
     return str_options.tmpPath + "/root" + path;
 }
 
+// cleanup function to clean cached files that are too old
+void gc_cache()
+{
+
+    while(true){
+
+        // lock the deque
+        file_to_delete file;
+        bool is_empty;
+        {
+            std::lock_guard<std::mutex> lock(deque_lock);
+            is_empty = cleanup.empty();
+            if(!is_empty)
+            {
+                file = cleanup.front();
+            }
+        }
+
+        //if deque is empty, skip
+        if(is_empty)
+        {
+            //run it every 1 second
+            usleep(1000);
+            continue;
+        }
+
+
+        //check if the closed time is old enough to delete
+        if((time(NULL) - file.closed_time) > file_cache_timeout_in_seconds)
```

Comment:
can we invert and reduce nesting?

if((time(NULL) - file.closed_time) < file_cache_timeout_in_seconds)
{
   usleep(1000);
   continue;
}

....